WalletWasabi/KeyManagement/KeyManager.cs

```diff
@@ -817,7 +817,7 @@ public void AssertNetworkOrClearBlockState(Network expectedNetwork)
 					BlockchainState.BlockStates.Clear();
 					ToFileNoBlockchainStateLock();
 
-					if (lastNetwork != null)
+					if (lastNetwork != null && expectedNetwork != null)
```

Comment:
`expectedNetwork` should not be null.

Reply:
I am using it to clear the current block state, by calling it with `expectedNetwork` as null. If this is not there then it will display a confusing log message.

Reply:
I know why you are doing that and I know it works. Yet it still smells. Let me elaborate on the larger picture here why I feel so strongly you should not do this: `AssertNetworkOrClearBlockState` was a hack by me in the first place. For some reason our state gets corrupted and I needed to hack this correcting function here. And now, you are hacking on top of my hack. It's a bad practice to use/rely on my hack in other parts of the code in the first place, but relying on my hack with another hack is really bad.